BZ2093243 Improve baremetal master node replacement documentation

GroceryBoyJr opened the pull request:
Problem in the description of recovering etcd that shows up in baremetal installations. The original document is found at:
https://docs.openshift.com/container-platform/4.8/backup_and_restore/control_plane_backup_and_restore/replacing-unhealthy-etcd-member.html

This PR adds a new section that works correctly for baremetal etcd replacement.

Version(s):
4.8+

Issue:
https://bugzilla.redhat.com/show_bug.cgi?id=2093243

Link to docs preview:
http://file.rdu.redhat.com/jbrigman/etcd-baremetal/backup_and_restore/control_plane_backup_and_restore/replacing-unhealthy-etcd-member.html#restore-replace-stopped-baremetal-etcd-member_replacing-unhealthy-etcd-member

Additional information:
The problem was found by customer Nokia and originally posed as a TELCO issue.

GroceryBoyJr commented on modules/restore-replace-stopped-baremetal-etcd-member.adoc:
```diff
@@ -0,0 +1,481 @@
+[source,terminal]
+----
+NAME        VERSION   AVAILABLE   PROGRESSING   DEGRADED   SINCE   MESSAGE
+baremetal   4.10.12   True        False         False      3d15h
```
This document fixes a problem found in 4.10.12 and identified by telco customer NOKIA. If I edit a number just to make it look like it was run on 4.11, that might raise questions about fidelity.

lshilin-r replied:
ok. May be u can use something like <version>?

GroceryBoyJr replied:
@lshilin-r May I use the version of OpenShift that you yourself tested on?

GroceryBoyJr replied:
Done

GroceryBoyJr commented on modules/restore-replace-stopped-baremetal-etcd-member.adoc:
```diff
@@ -0,0 +1,481 @@
+[source,terminal]
+----
+$ oc delete bmh -n openshift-machine-api vmaster-0
+$ oc delete machine -n openshift-machine-api kni1-master-0
```
We did! Great catch, @lshilin-r ! Thank you so much!

lshilin-r replied:
BTW, looks like this examples are coming from different setup (see the name of the machine). In addition, the name of bmh is very confusing, it looks like it comes from virtual simulation.

I would delete BMH first. To find the list of BMH the user need to run a command:
$ oc get bmh -n openshift-machine-api
The BMH to delete has a consumer name of the machine to delete

GroceryBoyJr replied:
Agreed, thank you. I am glad to add the "oc get bmh ... "

lshilin-r replied:
$ oc get bmh -n openshift-machine-api
NAME                 STATE                    CONSUMER                        ONLINE   ERROR   AGE
openshift-master-0   externally provisioned   ocp-edge-4zg89-master-0         true             3h43m
openshift-master-1   externally provisioned   ocp-edge-4zg89-master-1         true             3h43m
openshift-master-2   externally provisioned   ocp-edge-4zg89-master-2         true             3h43m
openshift-worker-0   provisioned              ocp-edge-4zg89-worker-0-8v2jj   true             3h43m
openshift-worker-1   provisioned              ocp-edge-4zg89-worker-0-jb5z9   true             3h43m

GroceryBoyJr replied:
done

GroceryBoyJr commented on modules/restore-replace-stopped-baremetal-etcd-member.adoc:
```diff
@@ -0,0 +1,481 @@
+spec:
+  automatedCleaningMode: disabled
+  bmc:
+    address: redfish-virtualmedia+http://192.168.124.113:8000/redfish/v1/Systems/f87eaf82-b32d-4291-ace6-b28677964e78
```
Thank you so much. Please tell me again, is the spelling of the third protocol "idrac-vistualmedia" or "idrac-visualmedia"?

lshilin-r replied:
idrac-vistualmedia

GroceryBoyJr replied:
and the "oc get bmh ... " will give the user that information, right?

GroceryBoyJr replied:
Done!

GroceryBoyJr commented on modules/restore-replace-stopped-baremetal-etcd-member.adoc:
```diff
@@ -0,0 +1,481 @@
+$ oc delete machine -n openshift-machine-api clustername-8qw5l-master-0 <1>
+----
+<1> Specify the name of the control plane machine for the unhealthy node.
+
```
Thank you so much @lshilin-r for this important information. I will work to revise the document and incorporate what you have provided.

GroceryBoyJr replied:
Done

GroceryBoyJr commented:
@lshilin-r Please discuss with me - I was going to ADD this procedure to the two existing procedures shown at http://file.rdu.redhat.com/jbrigman/etcd-baremetal/backup_and_restore/control_plane_backup_and_restore/restore-replace-stopped-baremetal-etcd-member.html 
Do you believe I should simply ADD the new procedure or REPLACE one of the two existing procedures?

lshilin-r commented:
> @lshilin-r Please discuss with me - I was going to ADD this procedure to the two existing procedures shown at http://file.rdu.redhat.com/jbrigman/etcd-baremetal/backup_and_restore/control_plane_backup_and_restore/restore-replace-stopped-baremetal-etcd-member.html Do you believe I should simply ADD the new procedure or REPLACE one of the two existing procedures?

I try to understand. Your document has the same title as https://docs.openshift.com/container-platform/4.8/backup_and_restore/control_plane_backup_and_restore/replacing-unhealthy-etcd-member.html#restore-replace-stopped-etcd-member_replacing-unhealthy-etcd-member and looks like a better version of the last.

What was the exact complain re: the old version?

GroceryBoyJr commented:
@lshilin-r the problem with the old version is that it did not work properly when performed on openshift running on bare metal, but it was written as though it did. I added a section that was missing. The name of mine is slightly different, I added the word "baremetal": restore-replace-stopped-baremetal-etcd-member_replacing-unhealthy-etcd-member. I copied it from the first procedure and added the missing pieces. I am glad to replace that first procedure with this one if you agree.

lshilin-r commented:
Ok, now I see the difference and understand the purpose of the new part. The old one is for not-baremetal cluster. The new is for baremetal cluster, so the part of bmh was added.
By this logic we need all 3 parts :)
I think, I'll run the procedure on real BM and provide some examples (e.g., output of `oc get bmh` command)

lshilin-r commented on modules/restore-replace-stopped-baremetal-etcd-member.adoc:
```diff
@@ -0,0 +1,481 @@
++
+[source,terminal]
+----
+$ oc get pods -n openshift-etcd | grep -v etcd-quorum-guard | grep etcd
```
$ oc get pods -n openshift-etcd | grep -v etcd-guard| grep etcd

GroceryBoyJr replied:
Done

lshilin-r commented on modules/restore-replace-stopped-baremetal-etcd-member.adoc:
```diff
@@ -0,0 +1,481 @@
+----
+etcd-ip-10-0-131-183.ec2.internal                3/3     Running     0          123m
+etcd-ip-10-0-164-97.ec2.internal                 3/3     Running     0          123m
+etcd-ip-10-0-154-204.ec2.internal                3/3     Running     0          124m
```
etcd-openshift-master-0                5/5     Running     0          105m
etcd-openshift-master-1                5/5     Running     0          107m
etcd-openshift-master-2                5/5     Running     0          103m

GroceryBoyJr replied:
Done

lshilin-r commented on modules/restore-replace-stopped-baremetal-etcd-member.adoc:
```diff
@@ -0,0 +1,481 @@
++
+[source,terminal]
+----
+$ oc rsh -n openshift-etcd etcd-ip-10-0-154-204.ec2.internal
```
$ oc rsh -n openshift-etcd etcd-openshift-master-0

GroceryBoyJr replied:
Done

lshilin-r commented on modules/restore-replace-stopped-baremetal-etcd-member.adoc:
```diff
@@ -0,0 +1,481 @@
+| 757b6793e2408b6c | started |  ip-10-0-164-97.ec2.internal |  https://10.0.164.97:2380 |  https://10.0.164.97:2379 |
+| ca8c2990a0aa29d1 | started | ip-10-0-154-204.ec2.internal | https://10.0.154.204:2380 | https://10.0.154.204:2379 |
++------------------+---------+------------------------------+---------------------------+---------------------------+
+----
```
+------------------+---------+--------------------+---------------------------+---------------------------+------------+
|        ID        | STATUS  |        NAME        |        PEER ADDRS         |       CLIENT ADDRS        | IS LEARNER |
+------------------+---------+--------------------+---------------------------+---------------------------+------------+
| 7a8197040a5126c8 | started | openshift-master-2 | https://10.46.29.131:2380/ | https://10.46.29.131:2379/ |      false |
| 8d5abe9669a39192 | started | openshift-master-1 | https://10.46.29.130:2380/ | https://10.46.29.130:2379/ |      false |
| cc3830a72fc357f9 | started | openshift-master-0 | https://10.46.29.129:2380/ | https://10.46.29.129:2379/ |      false |
+------------------+---------+--------------------+---------------------------+---------------------------+------------+

GroceryBoyJr replied:
Done

lshilin-r commented on modules/restore-replace-stopped-baremetal-etcd-member.adoc:
```diff
@@ -0,0 +1,481 @@
++
+[source,terminal]
+----
+sh-4.2# etcdctl member remove 6fc1e7c9db35841d
```
sh-4.4# etcdctl member remove 7a8197040a5126c8

GroceryBoyJr replied:
Done

lshilin-r commented on modules/restore-replace-stopped-baremetal-etcd-member.adoc:
```diff
@@ -0,0 +1,481 @@
+.Example output
+[source,terminal]
+----
+Member 6fc1e7c9db35841d removed from cluster ead669ce1fbfb346
```
Member 7a8197040a5126c8 removed from cluster b23536c33f2cdd1b

GroceryBoyJr replied:
Done

lshilin-r commented on modules/restore-replace-stopped-baremetal-etcd-member.adoc:
```diff
@@ -0,0 +1,481 @@
++
+Take note of the ID and the name of the unhealthy etcd member, because these values are needed later in the procedure. The `$ etcdctl endpoint health` command will list the removed member until the procedure of replacement is finished and a new member is added.
+
+.. Remove the unhealthy etcd member by providing the ID to the `etcdctl member remove` command:
```
This part should be moved to after bmh deletion

GroceryBoyJr replied:
@lshilin-r It appears the problem is with the sentence "The `$ etcdctl endpoint health` command will list the removed member until the procedure of replacement is finished and a new member is added.", is this correct? If so, move only this sentence to after bmh deletion?

GroceryBoyJr replied:
@lshilin-r Please forgive me, I've not moved this yet - I need more clarification.

lshilin-r replied:
No need to move it. I thought I mentioned it here, but may be did it on slack only

lshilin-r commented on modules/restore-replace-stopped-baremetal-etcd-member.adoc:
```diff
@@ -0,0 +1,481 @@
+Existing control plane `BareMetalHost` objects may have the `externallyProvisioned` flag set to `true` if they were provisioned by the {product-title} installation program.
+====
+
+.Prerequisites
```
shouldn't this part be the first

GroceryBoyJr replied:
Yes! I have moved. Thank you!

lshilin-r commented on modules/restore-replace-stopped-baremetal-etcd-member.adoc:
```diff
@@ -0,0 +1,481 @@
+      type: MachineCreation
+    instanceId: i-0fdb85790d76d0c3f
+    instanceState: stopped
+    kind: AWSMachineProviderStatus
```
status:
  addresses:
  - address: ""
    type: InternalIP
  - address: fe80::4adf:37ff:feb0:8aa1%ens1f1.373
    type: InternalIP
  - address: fe80::4adf:37ff:feb0:8aa1%ens1f1.371
    type: InternalIP
  - address: fe80::4adf:37ff:feb0:8aa1%ens1f1.341
    type: InternalIP
  - address: ""
    type: InternalIP
  - address: 10.46.29.131
    type: InternalIP
  - address: fe80::4adf:37ff:feb0:8aa1%ens1f1
    type: InternalIP
  - address: fe80::4adf:37ff:feb0:8aa0%ens1f0.342
    type: InternalIP
  - address: fe80::4adf:37ff:feb0:8aa1%ens1f1.372
    type: InternalIP
  - address: fe80::4adf:37ff:feb0:8aa1%ens1f1.370
    type: InternalIP
  - address: ""
    type: InternalIP
  - address: 16.1.15.2
    type: InternalIP
  - address: fe80::64f7:72ff:feec:f155%enp1s0f4u4
    type: InternalIP
  - address: openshift-master-2
    type: Hostname
  - address: openshift-master-2
    type: InternalDNS

GroceryBoyJr replied:
done

GroceryBoyJr replied:
@lshilin-r I used only a subset of this output - do you prefer I use ALL of it?

lshilin-r commented on modules/restore-replace-stopped-baremetal-etcd-member.adoc:
```diff
@@ -0,0 +1,481 @@
+kind: Machine
+metadata:
+  ...
+  name: clustername-8qw5l-master-3
```
name: ocp-edge-4zg89-master-3

GroceryBoyJr replied:
done

lshilin-r commented on modules/restore-replace-stopped-baremetal-etcd-member.adoc:
```diff
@@ -0,0 +1,481 @@
++
+[source,yaml]
+----
+  providerID: aws:///us-east-1a/i-0fdb85790d76d0c3f
```
providerID: baremetalhost:///openshift-machine-api/openshift-master-2/3354bdac-61d8-410f-be5b-6a395b056135

GroceryBoyJr replied:
Done

lshilin-r commented on modules/restore-replace-stopped-baremetal-etcd-member.adoc:
```diff
@@ -0,0 +1,481 @@
+[source,yaml]
+----
+  annotations:
+    machine.openshift.io/instance-state: running
```
machine.openshift.io/instance-state: externally provisioned

GroceryBoyJr replied:
done

lshilin-r commented on modules/restore-replace-stopped-baremetal-etcd-member.adoc:
```diff
@@ -0,0 +1,481 @@
+  resourceVersion: "13291"
+  uid: a282eb70-40a2-4e89-8009-d05dd420d31a
+----
+
```
Remove the `spec.conditions`,  `spec.lastUpdated`,  `spec.nodeRef` and `spec.phase`

  conditions:
  - lastTransitionTime: "2022-08-03T08:40:36Z"
    message: 'Drain operation currently blocked by: [{Name:EtcdQuorumOperator Owner:clusteroperator/etcd}]'
    reason: HookPresent
    severity: Warning
    status: "False"
    type: Drainable
  - lastTransitionTime: "2022-08-03T08:39:55Z"
    status: "True"
    type: InstanceExists
  - lastTransitionTime: "2022-08-03T08:36:37Z"
    status: "True"
    type: Terminable
  lastUpdated: "2022-08-03T08:40:36Z"
  nodeRef:
    kind: Node
    name: openshift-master-2
    uid: 788df282-6507-4ea2-9a43-24f237ccbc3c
  phase: Running

GroceryBoyJr replied:
done

lshilin-r commented on modules/restore-replace-stopped-baremetal-etcd-member.adoc:
```diff
@@ -0,0 +1,481 @@
+clustername-8qw5l-master-2                  Running   m4.xlarge   us-east-1   us-east-1c   3h37m   ip-10-0-164-97.ec2.internal    aws:///us-east-1c/i-02626f1dba9ed5bba   running
+clustername-8qw5l-worker-us-east-1a-wbtgd   Running   m4.large    us-east-1   us-east-1a   3h28m   ip-10-0-129-226.ec2.internal   aws:///us-east-1a/i-010ef6279b4662ced   running
+clustername-8qw5l-worker-us-east-1b-lrdxb   Running   m4.large    us-east-1   us-east-1b   3h28m   ip-10-0-144-248.ec2.internal   aws:///us-east-1b/i-0cb45ac45a166173b   running
+clustername-8qw5l-worker-us-east-1c-pkg26   Running   m4.large    us-east-1   us-east-1c   3h28m   ip-10-0-170-181.ec2.internal   aws:///us-east-1c/i-06861c00007751b0a   running
```
$ oc get machines -n openshift-machine-api -o wide
NAME                            PHASE     TYPE   REGION   ZONE   AGE     NODE                 PROVIDERID                                                                                       STATE
ocp-edge-4zg89-master-0         Running                          3h39m   openshift-master-0   baremetalhost:///openshift-machine-api/openshift-master-0/da1ebe11-3ff2-41c5-b099-0aa41222964e   externally provisioned
ocp-edge-4zg89-master-1         Running                          3h39m   openshift-master-1   baremetalhost:///openshift-machine-api/openshift-master-1/d9f9acbc-329c-475e-8d81-03b20280a3e1   externally provisioned
ocp-edge-4zg89-worker-0-8v2jj   Running                          3h13m   openshift-worker-0   baremetalhost:///openshift-machine-api/openshift-worker-0/3d685b81-7410-4bb3-80ec-13a31858241f   provisioned
ocp-edge-4zg89-worker-0-jb5z9   Running                          3h13m   openshift-worker-1   baremetalhost:///openshift-machine-api/openshift-worker-1/0fdae6eb-2066-4241-91dc-e7ea72ab13b9   provisioned

GroceryBoyJr replied:
Done

lshilin-r commented on modules/restore-replace-stopped-baremetal-etcd-member.adoc:
```diff
@@ -0,0 +1,481 @@
+baremetal   4.10.12   True        False         False      3d15h
+----
+
+. Remove the old `BareMetalHost` and `Machine` objects:
```
```suggestion
. Remove the old `BareMetalHost` object:
```

GroceryBoyJr replied:
Done

lshilin-r commented on modules/restore-replace-stopped-baremetal-etcd-member.adoc:
```diff
@@ -0,0 +1,481 @@
+[source,terminal]
+----
+$ oc delete bmh -n openshift-machine-api vmaster-0
+$ oc delete machine -n openshift-machine-api kni1-master-0
```
```suggestion
$ oc delete bmh openshift-master-2 -n openshift-machine-api
baremetalhost.metal3.io "openshift-master-2" deleted
```

lshilin-r replied:
please, don't forget to remove `delete machine`

GroceryBoyJr replied:
Done

lshilin-r commented on modules/restore-replace-stopped-baremetal-etcd-member.adoc:
```diff
@@ -0,0 +1,481 @@
+$ oc delete machine -n openshift-machine-api kni1-master-0
+----
++
+After you remove the `BareMetalHost` and `Machine` objects, then the `Machine` controller automatically deletes the `Node` object.
```
sometimes it takes a little long for machine controller to realize the bmh and machine deleted. In this case the node can be deleted by
$ oc delete node openshift-master-2
node "openshift-master-2" deleted

$ oc get nodes
NAME                 STATUS   ROLES    AGE     VERSION
openshift-master-0   Ready    master   3h24m   v1.24.0+9546431
openshift-master-1   Ready    master   3h24m   v1.24.0+9546431
openshift-worker-0   Ready    worker   176m    v1.24.0+9546431
openshift-worker-1   Ready    worker   176m    v1.24.0+9546431

GroceryBoyJr replied:
Done. Thank you for this wonderful addition to the document!

lshilin-r commented on modules/restore-replace-stopped-baremetal-etcd-member.adoc:
```diff
@@ -0,0 +1,481 @@
+  bootMode: UEFI
+  externallyProvisioned: false
+  hardwareProfile: unknown
+  online: true
```
apiVersion: v1
kind: Secret
metadata:
  name: openshift-master-2-bmc-secret
  namespace: openshift-machine-api
data:
  password: <password>
  username: <username>
type: Opaque
---
apiVersion: metal3.io/v1alpha1
kind: BareMetalHost
metadata:
  name: openshift-master-2
  namespace: openshift-machine-api
spec:
  automatedCleaningMode: disabled
  bmc:
    address: redfish-virtualmedia://10.46.61.18:443/redfish/v1/Systems/1
    credentialsName: openshift-master-2-bmc-secret
    disableCertificateVerification: true
  bootMACAddress: 48:df:37:b0:8a:a0
  bootMode: UEFI
  externallyProvisioned: false
  online: true
  rootDeviceHints:
    deviceName: /dev/sda
  userData:
    name: master-user-data-managed
    namespace: openshift-machine-api

GroceryBoyJr replied:
Done

GroceryBoyJr replied:
Also: added the Note:

lshilin-r commented on modules/restore-replace-stopped-baremetal-etcd-member.adoc:
```diff
@@ -0,0 +1,481 @@
+kni1-master-1  externally provisioned   ocp-hkw9p-master-1         true             4h53m
+kni1-master-2  externally provisioned   ocp-hkw9p-master-2         true             4h53m
+kni1-worker-0  provisioned              ocp-hkw9p-worker-0-ktmmx   true             4h53m
+kni1-worker-1  provisioned              ocp-hkw9p-worker-0-l2zmb   true             4h53m
```
on this stage the new BMH have no consumer yet. 
$ oc get bmh -n openshift-machine-api
NAME                 STATE                    CONSUMER                        ONLINE   ERROR   AGE
openshift-master-0   externally provisioned   ocp-edge-4zg89-master-0         true             4h6m
openshift-master-1   externally provisioned   ocp-edge-4zg89-master-1         true             4h6m
openshift-master-2   available                                               true             35m14s
openshift-worker-0   provisioned              ocp-edge-4zg89-worker-0-8v2jj   true             4h6m
openshift-worker-1   provisioned              ocp-edge-4zg89-worker-0-jb5z9   true             4h6m

GroceryBoyJr replied:
Done

lshilin-r commented on modules/restore-replace-stopped-baremetal-etcd-member.adoc:
```diff
@@ -0,0 +1,481 @@
+<1> The new machine, `clustername-8qw5l-master-3` is being created and is ready once the phase changes from `Provisioning` to `Running`.
++
+It might take a few minutes for the new machine to be created. The etcd cluster Operator will automatically sync when the machine or node returns to a healthy state.
+
```
$ oc get bmh -n openshift-machine-api
NAME                 STATE                    CONSUMER                        ONLINE   ERROR   AGE
openshift-master-0   externally provisioned   ocp-edge-4zg89-master-0         true             4h48m
openshift-master-1   externally provisioned   ocp-edge-4zg89-master-1         true             4h48m
openshift-master-2   provisioned              ocp-edge-4zg89-master-3         true             47m
openshift-worker-0   provisioned              ocp-edge-4zg89-worker-0-8v2jj   true             4h48m
openshift-worker-1   provisioned              ocp-edge-4zg89-worker-0-jb5z9   true             4h48m
[kni@ocp-edge06 ~]$ oc get machine -n openshift-machine-api
NAME                            PHASE     TYPE   REGION   ZONE   AGE
ocp-edge-4zg89-master-0         Running                          4h49m
ocp-edge-4zg89-master-1         Running                          4h48m
ocp-edge-4zg89-master-3         Running                          25m
ocp-edge-4zg89-worker-0-8v2jj   Running                          4h23m
ocp-edge-4zg89-worker-0-jb5z9   Running                          4h23m
[kni@ocp-edge06 ~]$ oc get nodes
NAME                 STATUS   ROLES    AGE     VERSION
openshift-master-0   Ready    master   4h26m   v1.24.0+9546431
openshift-master-1   Ready    master   4h26m   v1.24.0+9546431
openshift-master-2   Ready    master   12m     v1.24.0+9546431
openshift-worker-0   Ready    worker   3h58m   v1.24.0+9546431
openshift-worker-1   Ready    worker   3h58m   v1.24.0+9546431

lshilin-r replied:
did you decide not to give this examples of the expected cluster state?

GroceryBoyJr replied:
I'm glad to do so but formatting of the output text is challenging.....

GroceryBoyJr replied:
Done

lshilin-r replied:
It looks like it added not to the right place (see my previous comment)

lshilin-r commented on modules/restore-replace-stopped-baremetal-etcd-member.adoc:
```diff
@@ -0,0 +1,481 @@
++
+[source,terminal]
+----
+$ oc get pods -n openshift-etcd | grep -v etcd-quorum-guard | grep etcd
```
$ oc get pods -n openshift-etcd | grep -v etcd-guard| grep etcd

GroceryBoyJr replied:
done

lshilin-r commented on modules/restore-replace-stopped-baremetal-etcd-member.adoc:
```diff
@@ -0,0 +1,481 @@
+etcd-ip-10-0-133-53.ec2.internal                 3/3     Running     0          7m49s
+etcd-ip-10-0-164-97.ec2.internal                 3/3     Running     0          123m
+etcd-ip-10-0-154-204.ec2.internal                3/3     Running     0          124m
+----
```
etcd-openshift-master-0                 5/5     Running     0          9m36s
etcd-openshift-master-1                 5/5     Running     0          11m
etcd-openshift-master-2                 5/5     Running     0          7m36s

GroceryBoyJr replied:
Done

lshilin-r commented on modules/restore-replace-stopped-baremetal-etcd-member.adoc:
```diff
@@ -0,0 +1,481 @@
++
+[source,terminal]
+----
+$ oc rsh -n openshift-etcd etcd-ip-10-0-154-204.ec2.internal
```
$ oc rsh -n openshift-etcd etcd-openshift-master-0

GroceryBoyJr replied:
Done

lshilin-r commented on modules/restore-replace-stopped-baremetal-etcd-member.adoc:
```diff
@@ -0,0 +1,481 @@
+| 5eb0d6b8ca24730c | started |  ip-10-0-133-53.ec2.internal |  https://10.0.133.53:2380 |  https://10.0.133.53:2379 |
+| 757b6793e2408b6c | started |  ip-10-0-164-97.ec2.internal |  https://10.0.164.97:2380 |  https://10.0.164.97:2379 |
+| ca8c2990a0aa29d1 | started | ip-10-0-154-204.ec2.internal | https://10.0.154.204:2380 | https://10.0.154.204:2379 |
++------------------+---------+------------------------------+---------------------------+---------------------------+
```
+------------------+---------+--------------------+---------------------------+---------------------------+------------+
|        ID        | STATUS  |        NAME        |        PEER ADDRS         |       CLIENT ADDRS        | IS LEARNER |
+------------------+---------+--------------------+---------------------------+---------------------------+------------+
| 8d5abe9669a39192 | started | openshift-master-1 | https://10.46.29.130:2380 | https://10.46.29.130:2379 |      false |
| cb1ce22c4574be06 | started | openshift-master-2 | https://10.46.29.131:2380 | https://10.46.29.131:2379 |      false |
| cc3830a72fc357f9 | started | openshift-master-0 | https://10.46.29.129:2380 | https://10.46.29.129:2379 |      false |
+------------------+---------+--------------------+---------------------------+---------------------------+------------+

GroceryBoyJr replied:
Done

lshilin-r commented on modules/restore-replace-stopped-baremetal-etcd-member.adoc:
```diff
@@ -0,0 +1,481 @@
+[WARNING]
+====
+Be sure to remove the correct etcd member; removing a good etcd member might lead to quorum loss.
+====
```
the following verification can be added:
Validate that all etcd members are healthy 
```
sh-4.4# etcdctl endpoint health --cluster 
https://10.46.29.130:2379 is healthy: successfully committed proposal: took = 8.973065ms
https://10.46.29.129:2379 is healthy: successfully committed proposal: took = 11.559829ms
https://10.46.29.131:2379 is healthy: successfully committed proposal: took = 11.665203ms

```



Validate that all nodes are at the latest revision
```
$ oc get etcd -o=jsonpath='{range.items[0].status.conditions[?(@.type=="NodeInstallerProgressing")]}{.reason}{"\n"}{.message}{"\n"}'
AllNodesAtLatestRevision
```

GroceryBoyJr replied:
Done, thank you!

GroceryBoyJr commented:
Hello @lshilin-r I have corrected those three items we last discussed: 
1. Numbering
2. Names of bmh/machine/node objects
3. move "Ensure that the bare metal operator is available:" up. 
Please, can you take a look at the work? I have some small style changes to make but nothing that affects the meaning.

lshilin-r commented on modules/restore-replace-stopped-baremetal-etcd-member.adoc:
```diff
@@ -0,0 +1,579 @@
+----
+etcd-openshift-master-10                5/5     Running     11          3h56m   192.168.10.9   openshift-master-10   <none>           <none>
+etcd-openshift-master-11                5/5     Running     0          3h54m   192.168.10.10   openshift-master-11   <none>           <none>
+etcd-openshift-master-12                5/5     Running     0          3h58m   192.168.10.11   openshift-master-12   <none>           <none>
```
why 10, 11, 12?

GroceryBoyJr replied:
The prior ones were in sequence, 0, 1, 2. I wanted to change the text somewhat, so to keep the sequence "correct", I chose simply to add 10 to each name.

GroceryBoyJr commented:
@lshilin-r I'm reviewing the edits to make sure I did everything you requested. I found one this morning that I missed. Thank you for your feedback!

lshilin-r commented on modules/restore-replace-stopped-baremetal-etcd-member.adoc:
```diff
@@ -0,0 +1,579 @@
+----
+NAME                            PHASE     TYPE   REGION   ZONE   AGE     NODE                 PROVIDERID                                                                                       STATE
+ocp-edge-4zg89-master-10         Running                          3h11m   openshift-master-0   baremetalhost:///openshift-machine-api/openshift-master-10/da1ebe11-3ff2-41c5-b099-0aa41222964e   externally provisioned
+ocp-edge-4zg89-master-11         Running                          3h11m   openshift-master-1   baremetalhost:///openshift-machine-api/openshift-master-1/d9f9acbc-329c-475e-8d81-03b20280a3e1   externally provisioned
```
please, change here as well to 0 and 1

GroceryBoyJr replied:
Thank you! Those have "tab" instead of "space" so my search missed those.

GroceryBoyJr replied:
Done! Also found another glitch and corrected it, so thank you, twice!

lshilin-r replied:
why I still see 10 and 11?

GroceryBoyJr commented on modules/restore-replace-stopped-baremetal-etcd-member.adoc:
```diff
@@ -0,0 +1,579 @@
+----
+$ oc delete node openshift-master-12
+
+node "openshift-master-12" deleted
```
Thank You! Done, please stand by.

emmahone reviewed (approved):
/lgtm

lshilin-r commented on modules/restore-replace-stopped-baremetal-etcd-member.adoc:
```diff
@@ -0,0 +1,592 @@
+[source,terminal]
+----
+NAME                            PHASE     TYPE   REGION   ZONE   AGE     NODE                 PROVIDERID                                                                                       STATE
+ocp-edge-4zg89-master-0         Running                          3h11m   openshift-master-0   baremetalhost:///openshift-machine-api/openshift-master-10/da1ebe11-3ff2-41c5-b099-0aa41222964e   externally provisioned <1>
```
```suggestion
examplecluster-control-plane-0         Running                          3h11m   openshift-master-0   baremetalhost:///openshift-machine-api/openshift-master-1/da1ebe11-3ff2-41c5-b099-0aa41222964e   externally provisioned <1>
```

GroceryBoyJr replied:
Done, thank you!

lshilin-r commented on modules/restore-replace-stopped-baremetal-etcd-member.adoc:
```diff
@@ -0,0 +1,592 @@
++
+[source,yaml]
+----
+  providerID: baremetalhost:///openshift-machine-api/openshift-master-12/3354bdac-61d8-410f-be5b-6a395b056135
```
```suggestion
  providerID: baremetalhost:///openshift-machine-api/openshift-control-plane-2/3354bdac-61d8-410f-be5b-6a395b056135
```

GroceryBoyJr replied:
Done, thank you!

lshilin-r commented on modules/restore-replace-stopped-baremetal-etcd-member.adoc:
```diff
@@ -0,0 +1,592 @@
+[source,terminal]
+----
+NAME                            PHASE     TYPE   REGION   ZONE   AGE     NODE                 PROVIDERID                                                                                       STATE
+ocp-edge-4zg89-master-10         Running                          3h11m   openshift-master-0   baremetalhost:///openshift-machine-api/openshift-master-10/da1ebe11-3ff2-41c5-b099-0aa41222964e   externally provisioned
```
```suggestion
clusterexaqmple-control-plane-0         Running                          3h11m   openshift-control-plane-0   baremetalhost:///openshift-machine-api/openshift-control-plane-0/da1ebe11-3ff2-41c5-b099-0aa41222964e   externally provisioned
```

lshilin-r replied:
the same for following rows

GroceryBoyJr replied:
Done, thank you!

lshilin-r commented on modules/restore-replace-stopped-baremetal-etcd-member.adoc:
```diff
@@ -0,0 +1,592 @@
+----
+$ oc delete bmh openshift-master-2 -n openshift-machine-api
+
+baremetalhost.metal3.io "openshift-master-12" deleted
```
12 -> 2

GroceryBoyJr replied:
Done, thank you!

lshilin-r commented on modules/restore-replace-stopped-baremetal-etcd-member.adoc:
```diff
@@ -0,0 +1,592 @@
++
+[source,terminal]
+----
+$ oc delete node openshift-master-12
```
ditto

GroceryBoyJr replied:
Done, thank you!

lshilin-r commented on modules/restore-replace-stopped-baremetal-etcd-member.adoc:
```diff
@@ -0,0 +1,592 @@
+----
+$ oc delete node openshift-master-12
+
+node "openshift-master-12" deleted
```
ditto

GroceryBoyJr replied:
Done, thank you!

lshilin-r commented on modules/restore-replace-stopped-baremetal-etcd-member.adoc:
```diff
@@ -0,0 +1,592 @@
+[source,terminal]
+----
+NAME                            PHASE     TYPE   REGION   ZONE   AGE     NODE                 PROVIDERID                                                                                       STATE
+ocp-edge-4zg89-master-0         Running                          3h11m   openshift-master-0   baremetalhost:///openshift-machine-api/openshift-master-10/da1ebe11-3ff2-41c5-b099-0aa41222964e   externally provisioned <1>
```
as mentioned above

GroceryBoyJr replied:
Done, thank you!

lshilin-r commented on modules/restore-replace-stopped-baremetal-etcd-member.adoc:
```diff
@@ -0,0 +1,592 @@
++
+[source,terminal]
+----
+$ oc rsh -n openshift-etcd etcd-openshift-master-10
```
10 -> 0

GroceryBoyJr replied:
Done. Thank you!

lshilin-r commented on modules/restore-replace-stopped-baremetal-etcd-member.adoc:
```diff
@@ -0,0 +1,592 @@
+openshift-master-2 Ready master 12m v1.24.0+9546431
+openshift-worker-0 Ready worker 3h58m v1.24.0+9546431
+openshift-worker-1 Ready worker 3h58m v1.24.0+9546431
+----
```
Please, delete rows 458-476. On this stage no machine and no node are created yet, only bmh is added in the state available

GroceryBoyJr replied:
Done, thank you!

GroceryBoyJr commented on modules/restore-replace-stopped-baremetal-etcd-member.adoc:
```diff
@@ -0,0 +1,592 @@
+NAME STATE CONSUMER ONLINE ERROR AGE
+openshift-master-0 externally provisioned ocp-edge-4zg89-master-0 true 4h48m
+openshift-master-1 externally provisioned ocp-edge-4zg89-master-1 true 4h48m
+openshift-master-2 provisioned ocp-edge-4zg89-master-3 true 47m
```
Done. Thank you!

GroceryBoyJr commented:
@lshilin-r stand by, generating an update to the preview now....process will take another 15 min or so from this message.

lshilin-r commented on modules/restore-replace-stopped-baremetal-etcd-member.adoc:
```diff
@@ -0,0 +1,574 @@
+examplecluster-control-plane-2         Running                          3h11m   openshift-control-plane-2   baremetalhost:///openshift-machine-api/openshift-control-plane-2/3354bdac-61d8-410f-be5b-6a395b056135   externally provisioned
+examplecluster-compute-0   Running                          165m    openshift-compute-0   baremetalhost:///openshift-machine-api/openshift-compute-0/3d685b81-7410-4bb3-80ec-13a31858241f   provisioned
+examplecluster-compute-1   Running                          165m    openshift-compute-1   baremetalhost:///openshift-machine-api/openshift-compute-1/0fdae6eb-2066-4241-91dc-e7ea72ab13b9   provisioned
+----
```
2 last lines problem with tabs

lshilin-r commented on modules/restore-replace-stopped-baremetal-etcd-member.adoc:
```diff
@@ -0,0 +1,574 @@
++
+[source,terminal]
+----
+$ oc get machine clustername-8qw5l-master-2 \ <1>
```
```suggestion
$ oc get machine examplecluster-control-plane-2 \ <1>
```

GroceryBoyJr replied:
Done! Thank you!

lshilin-r commented on modules/restore-replace-stopped-baremetal-etcd-member.adoc:
```diff
@@ -0,0 +1,574 @@
+
+. Change the `metadata.name` field to a new name.
++
+It is recommended to keep the same base name as the old machine and change the ending number to the next available number. In this example, `clustername-8qw5l-master-0` is changed to `clustername-8qw5l-master-2`.
```
```suggestion
It is recommended to keep the same base name as the old machine and change the ending number to the next available number. In this example, `examplecluster-control-plane-2` is changed to `examplecluster-control-plane-3`.
```

GroceryBoyJr replied:
Done, thank you!

lshilin-r commented on modules/restore-replace-stopped-baremetal-etcd-member.adoc:
```diff
@@ -0,0 +1,574 @@
++
+[source,terminal]
+----
+$ oc delete machine -n openshift-machine-api clustername-8qw5l-master-2 <1>
```
naming

GroceryBoyJr replied:
Done, thank you!

lshilin-r commented on modules/restore-replace-stopped-baremetal-etcd-member.adoc:
```diff
@@ -0,0 +1,574 @@
++
+[source,terminal]
+----
+$ oc edit machine -n openshift-machine-api clustername-8qw5l-master-0
```
naming. And we are deleting control-plane-2, so finalizer should be deleted for him :)

lshilin-r commented on modules/restore-replace-stopped-baremetal-etcd-member.adoc:
```diff
@@ -0,0 +1,574 @@
++
+[source,terminal]
+----
+machine.machine.openshift.io/clustername8qw5l-master-0 edited
```
ditto

GroceryBoyJr replied:
Done, thank you!

lshilin-r commented on modules/restore-replace-stopped-baremetal-etcd-member.adoc:
```diff
@@ -0,0 +1,574 @@
+examplecluster-control-plane-2         Running                          3h11m   openshift-control-plane-2   baremetalhost:///openshift-machine-api/openshift-control-plane-2/3354bdac-61d8-410f-be5b-6a395b056135   externally provisioned
+examplecluster-compute-0   Running                          165m    openshift-compute-0   baremetalhost:///openshift-machine-api/openshift-compute-0/3d685b81-7410-4bb3-80ec-13a31858241f   provisioned
+examplecluster-compute-1   Running                          165m    openshift-compute-1   baremetalhost:///openshift-machine-api/openshift-compute-1/0fdae6eb-2066-4241-91dc-e7ea72ab13b9   provisioned 
+----
```
tabbing for 2 last lines

GroceryBoyJr replied:
Done, thank you!

lshilin-r commented:
/LGTM

jldohmann commented on modules/restore-replace-stopped-baremetal-etcd-member.adoc:
```diff
@@ -0,0 +1,612 @@
+After you remove the member, the cluster might be unreachable for a short time while the remaining etcd instances reboot.
+====
+
+. Remove the old secrets for the unhealthy etcd member that was removed.
```
The substeps in this part of the procedure need the "by running the following command" or [similar explicit instructions](https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/doc_guidelines.adoc#code-blocks-command-syntax-and-example-output) appended to it

GroceryBoyJr replied:
Done - good point!

jldohmann commented on modules/restore-replace-stopped-baremetal-etcd-member.adoc:
```diff
@@ -0,0 +1,612 @@
+
+. Delete the control plane machine.
++
+If you are running installer-provisioned infrastructure, or you used the Machine API to create your machines, follow these steps. Otherwise, you must create the new master using the same method that was used to originally create it.
```
Similar to an earlier comment: is there a way we can avoid using the term "master"?

jldohmann commented on modules/restore-replace-stopped-baremetal-etcd-member.adoc:
```diff
@@ -0,0 +1,612 @@
+    -o yaml \
+    > new-master-machine.yaml
+----
+Specify the name of the control plane machine for the unhealthy node.
```
Did you mean for this to be the callout? E.g.
```suggestion
<1> Specify the name of the control plane machine for the unhealthy node.
```

jldohmann commented on modules/restore-replace-stopped-baremetal-etcd-member.adoc:
```diff
@@ -0,0 +1,612 @@
+  ...
+----
+
+.. Remove the `spec.providerID` field:
```
Are these substeps? They seem disconnected from the originating step. Maybe a way to approach this whole step is to have the parent step be "Change or remove the following fields", and have the substep start with what is currently step 4. What do you think?

jldohmann replied:
@GroceryBoyJr A note sounds like a good idea! 👍

jldohmann commented on modules/restore-replace-stopped-baremetal-etcd-member.adoc:
```diff
@@ -0,0 +1,612 @@
+  generation: 2
+----
+
+.. Remove the `spec.conditions`, `spec.lastUpdated`, `spec.nodeRef` and `spec.phase' fields:
```
```suggestion
.. Remove the `spec.conditions`, `spec.lastUpdated`, `spec.nodeRef` and `spec.phase` fields:
```

GroceryBoyJr replied:
This is where I found a "hidden backtick". I THINK it's a vscode/codium thing (trying to "close" the backticks) but I'm not sure at this point. Will take a look at the file with vim to double-check.

jldohmann commented on modules/restore-replace-stopped-baremetal-etcd-member.adoc:
```diff
@@ -0,0 +1,612 @@
+phase: Running
+----
+
+. Ensure that the Bare Metal Operator is available:
```
```suggestion
. Ensure that the Bare Metal Operator is available by running the following command:
```

GroceryBoyJr replied:
Agreed, thank you.

jldohmann commented on modules/restore-replace-stopped-baremetal-etcd-member.adoc:
```diff
@@ -0,0 +1,612 @@
+baremetal   4.11.3    True        False         False      3d15h
+----
+
+. Delete the machine of the unhealthy member:
```
```suggestion
. Delete the machine of the unhealthy member by running the following command:
```

GroceryBoyJr replied:
I went with "using this command" to try to be a little less repetitive, but completely agree with your suggestion!

jldohmann commented on modules/restore-replace-stopped-baremetal-etcd-member.adoc:
```diff
@@ -0,0 +1,612 @@
+$ oc delete machine -n openshift-machine-api examplecluster-control-plane-2 <1>
+----
++
+If deletion of the machine is delayed for any reason or the command is obstructed and delayed, you can force deletion by removing the machine object finalizer field. 
```
Did you mean for this to be a callout? E.g.
```suggestion
<1> If deletion of the machine is delayed for any reason or the command is obstructed and delayed, you can force deletion by removing the machine object finalizer field. 
```

jldohmann commented on modules/restore-replace-stopped-baremetal-etcd-member.adoc:
```diff
@@ -0,0 +1,612 @@
++
+[IMPORTANT]
+====
+Do not interrupt machine deletion with `control-c`. You must allow the command to proceed to completion. Open a new terminal window to edit and delete the finalizer fields.
```
ISG nit:
```suggestion
Do not interrupt machine deletion by pressing `Ctrl+c`. You must allow the command to proceed to completion. Open a new terminal window to edit and delete the finalizer fields.
```

GroceryBoyJr replied:
Oh dang! Thank you!

jldohmann commented on modules/restore-replace-stopped-baremetal-etcd-member.adoc:
```diff
@@ -0,0 +1,612 @@
+Do not interrupt machine deletion with `control-c`. You must allow the command to proceed to completion. Open a new terminal window to edit and delete the finalizer fields.
+====
++
+[source,terminal]
```
Is this code block a duplicate?

GroceryBoyJr replied:
No, the first one is the primary command, the second one is a follow-on command to delete the finalizer fields.

jldohmann commented on modules/restore-replace-stopped-baremetal-etcd-member.adoc:
```diff
@@ -0,0 +1,612 @@
+- machine.machine.openshift.io
+----
++
+and save the change. 
```
```suggestion
Save your changes: 
```

GroceryBoyJr replied:
Brilliant, thank you.

jldohmann commented on modules/restore-replace-stopped-baremetal-etcd-member.adoc:
```diff
@@ -0,0 +1,612 @@
+machine.machine.openshift.io/examplecluster-control-plane-2 edited
+----
++
+.. Verify the machine was deleted:
```
```suggestion
.. Verify the machine was deleted by running the following command:
```

GroceryBoyJr replied:
Yes! Done.

jldohmann commented on modules/restore-replace-stopped-baremetal-etcd-member.adoc:
```diff
@@ -0,0 +1,612 @@
+examplecluster-compute-1          Running                          165m    openshift-compute-1         baremetalhost:///openshift-machine-api/openshift-compute-1/0fdae6eb-2066-4241-91dc-e7ea72ab13b9         provisioned
+----
+
+. Remove the old `BareMetalHost` object:
```
```suggestion
. Remove the old `BareMetalHost` object by running the following command:
```

GroceryBoyJr replied:
Yes! Varied slightly to avoid repetition.

jldohmann commented on modules/restore-replace-stopped-baremetal-etcd-member.adoc:
```diff
@@ -0,0 +1,612 @@
+----
+$ oc delete bmh openshift-control-plane-2 -n openshift-machine-api
+
+baremetalhost.metal3.io "openshift-control-plane-2" deleted
```
I think this should be separated into an example output

GroceryBoyJr replied:
Agreed. Done!

jldohmann commented on modules/restore-replace-stopped-baremetal-etcd-member.adoc:
```diff
@@ -0,0 +1,612 @@
++
+After you remove the `BareMetalHost` and `Machine` objects, then the `Machine` controller automatically deletes the `Node` object.
++
+If, after deletion of bmh, the machine node requires excessive time for deletion, the machine node can be deleted using: 
```
```suggestion
If, after deletion of the `BareMetalHost` object, the machine node requires excessive time for deletion, the machine node can be deleted using: 
```

GroceryBoyJr replied:
Thank you, yes.

jldohmann commented on modules/restore-replace-stopped-baremetal-etcd-member.adoc:
```diff
@@ -0,0 +1,612 @@
+EOF
+----
++
+Note: username and password can be found from the other bmh's secrets. The protocol to use in bmc:address: can be taken from other bmh objects.
```
Is "bmh" a commonly understood acronym? If not, then suggestion to expand it out to bare metal host:
```suggestion
[NOTE]
====
The username and password can be found from the other bare metal host's secrets. The protocol to use in the `bmc.address` field can be taken from other Bare Metal Host objects.
====
```

GroceryBoyJr replied:
It is, but it's a specific modifier used in the command itself. Pulling that term out into the descriptions may affect readability to the point that I think I should go with your changes.  Also, SERIOUSLY agreed with you about changing the Note: to [NOTE]. I was afraid of repetition here but I think the change is a big improvement in consistency.

jldohmann commented on modules/restore-replace-stopped-baremetal-etcd-member.adoc:
```diff
@@ -0,0 +1,612 @@
+openshift-compute-1       provisioned            examplecluster-compute-1       true         4h48m
+----
++
+.. Create the new control plane machine using the `new-master-machine.yaml` file:
```
These substeps seem disconnected from the parent step. Should they be their own steps?

GroceryBoyJr replied:
Hmmm. i'm pondering that. I wanted the a,b,c,d to be "components of the 9th step". But that might be a bit complex? I'm thinking on it.....

jldohmann commented on modules/restore-replace-stopped-baremetal-etcd-member.adoc:
```diff
@@ -0,0 +1,612 @@
+It should take a few minutes for the new machine to be created. The etcd cluster Operator will automatically sync when the machine or node returns to a healthy state.
+
+
+.. Verify that bmh becomes provisioned and no error reported:
```
```suggestion
.. Verify that the bare metal host becomes provisioned and no error reported by running the following command:
```

GroceryBoyJr replied:
Done, thank you!

jldohmann commented on modules/restore-replace-stopped-baremetal-etcd-member.adoc:
```diff
@@ -0,0 +1,612 @@
+openshift-compute-1       provisioned            examplecluster-compute-1       true         4h48m
+----
+
+.. The new node is added and in ready state:
```
Is this a verification step?
```suggestion
.. Verify that the new node is added and in a ready state by running the following command:
```

GroceryBoyJr replied:
Done, YES! Thank You!

jldohmann commented on modules/restore-replace-stopped-baremetal-etcd-member.adoc:
```diff
@@ -0,0 +1,612 @@
++
+.To verify there are exactly three etcd members, connect to the running etcd container, passing in the name of a pod that was not on the affected node:
++
+.In a terminal that has access to the cluster as a `cluster-admin` user, run the following command:
```
```suggestion
. In a terminal that has access to the cluster as a `cluster-admin` user, run the following command:
```

GroceryBoyJr replied:
What...the...hale. I have corrected this. It was a small paragraph of direction that had gotten munged up with too many "+" signs....THANK YOU SO MUCH.

jldohmann commented on modules/restore-replace-stopped-baremetal-etcd-member.adoc:
```diff
@@ -0,0 +1,612 @@
++------------------+---------+--------------------+---------------------------+---------------------------+-----------------+
+----
++
+If the output from the previous command lists more than three etcd members, you must carefully remove the unwanted member.
```
Suggestion to turn this into a note:
```suggestion
[NOTE]
====
If the output from the previous command lists more than three etcd members, you must carefully remove the unwanted member.
====
```

GroceryBoyJr replied:
agreed! Thank you!

jldohmann commented on modules/restore-replace-stopped-baremetal-etcd-member.adoc:
```diff
@@ -0,0 +1,612 @@
+https://192.168.10.11:2379 is healthy: successfully committed proposal: took = 11.665203ms
+----
++
+. Validate that all nodes are at the latest revision
```
```suggestion
. Validate that all nodes are at the latest revision by running the following command:
```

GroceryBoyJr replied:
Butofcourse! Thank You!

jldohmann commented on modules/restore-replace-stopped-baremetal-etcd-member.adoc:
```diff
@@ -0,0 +1,612 @@
+If the output from the previous command lists more than three etcd members, you must carefully remove the unwanted member.
++
+
+. Verify that all etcd members are healthy:
```
```suggestion
. Verify that all etcd members are healthy by running the following command:
```

GroceryBoyJr replied:
WHO DOES THAT?!?! nOt mE of cOuRSe!

jldohmann commented on modules/restore-replace-stopped-baremetal-etcd-member.adoc:
```diff
@@ -0,0 +1,612 @@
++
+[source,terminal]
+----
+# edctl endpoint health --cluster
```
Did you mean `etcdctl` here?

GroceryBoyJr replied:
Fatigue. That's my only reasonable answer here. Just fatigue. DOH!

jldohmann reviewed (commented):
Mostly looking good! Minor nits and suggestions below

GroceryBoyJr commented on modules/restore-replace-stopped-baremetal-etcd-member.adoc:
```diff
@@ -0,0 +1,612 @@
+
+This procedure details the steps to replace a bare metal etcd member that is unhealthy either because the machine is not running or because the node is not ready.
+
+If you are running installer-provisioned infrastructure or you used the Machine API to create your machines, follow these steps. Otherwise you must create the new master using the same method that was used to originally create it. 
```
Excellent catch, thank you, yes!

GroceryBoyJr commented on modules/restore-replace-stopped-baremetal-etcd-member.adoc:
```diff
@@ -0,0 +1,612 @@
++
+[IMPORTANT]
+====
+It is important to take an etcd backup before performing this procedure so that your cluster can be restored if you encounter any issues.
```
VERY glad to make that change!

GroceryBoyJr commented on modules/restore-replace-stopped-baremetal-etcd-member.adoc:
```diff
@@ -0,0 +1,612 @@
++------------------+---------+--------------------+---------------------------+---------------------------+---------------------+
+----
++
+Take note of the ID and the name of the unhealthy etcd member, because these values are required later in the procedure. The `$ etcdctl endpoint health` command will list the removed member until the replacement procedure is completed and the new member is added.
```
Very good catch - that was a cut/paste error. Thank you!

mburke5678 commented:
/cherrypick enterprise-4.8

mburke5678 commented:
/cherrypick enterprise-4.9

mburke5678 commented:
/cherrypick enterprise-4.10

mburke5678 commented:
/cherrypick enterprise-4.11

mburke5678 commented:
/cherrypick enterprise-4.12

mburke5678 commented:
/cherrypick enterprise-4.8
